src/selectors/per-thread/markers.js

```diff
@@ -605,6 +610,110 @@ export function getMarkerSelectorsPerThread(
     }
   );
 
+  type MarkerTrackSelectors = $ReturnType<typeof _createMarkerTrackSelectors>;
+  const _markerTrackSelectors = {};
+  const getMarkerTrackSelectors = (
+    markerSchema: MarkerSchema,
+    markerName: IndexIntoStringTable
+  ): MarkerTrackSelectors => {
+    const name = markerSchema.name;
+    if (!_markerTrackSelectors[name]) {
+      _markerTrackSelectors[name] = {};
+    }
+    if (!_markerTrackSelectors[name][markerName]) {
+      _markerTrackSelectors[name][markerName] = _createMarkerTrackSelectors(
+        markerSchema,
+        markerName
+      );
+    }
+    return _markerTrackSelectors[name][markerName];
+  };
+
+  /**
+   * This function creates selectors for each of the trackable markers of a thread. The type
+   * signature of each selector is defined in the function body, and inferred in the return
+   * type of the function.
+   */
+  function _createMarkerTrackSelectors(
+    markerSchema: MarkerSchema,
+    markerName: IndexIntoStringTable
+  ) {
+    const getCollectedCustomMarkerSamples: Selector<CollectedCustomMarkerSamples> =
+      createSelector(
+        getFullMarkerList,
+        threadSelectors.getStringTable,
+        (fullMarkerList, stringTable) => {
+          if (markerSchema.graphs === undefined) {
+            throw new Error(
+              `No graphs for marker ${markerName}. This shouldn't happen.`
+            );
+          }
+          const indexes = [];
+          const markers = [];
```

Comment:
I wonder why we need both marker indexes and markers? I guess we can have markers from marker indexes. Is it more convenient this way or is there another reason?

Reply:
Thanks for doing it. I think it's good to do it this way. I don't think we need to keep markers on top of the indexes as well, and a lot of our code returns marker indexes so it's good to be consistent.